### ISSUE TITLE
Salt Minion Memory Leak

### ISSUE THREAD

**Issue by HeathNaylor** [saltstack/salt#32481]
I have a memory leak on my Salt Minions, the only job running on them is the Salt Mine Interval. I currently have that set to 1 minute. The minions were setup via Salt-Cloud

### Setup
```
stage-web:
  extends: base
  minion:
    master: my.site.com
    master_port: 4508
    autoload_dynamic_modules: true
    mine_interval: 1
```

### Steps to Reproduce Issue
I start the minion, wait a few days and pretty soon I am up to 250+MB in RES memory
![image](https://cloud.githubusercontent.com/assets/10814256/14435722/796dfd9e-ffd6-11e5-81e9-2d134698064f.png)

No real logs, just the mine job.
```
2016-04-11 16:16:37,665 [salt.utils.schedule                                  ][INFO    ][14206] Running scheduled job: __mine_interval
```
### Versions Report
```
Salt Version:
           Salt: 2015.8.5

Dependency Versions:
         Jinja2: 2.7.2
       M2Crypto: Not Installed
           Mako: 0.9.1
         PyYAML: 3.10
          PyZMQ: 14.0.1
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
           RAET: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.4
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: Not Installed
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.3.0
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
   python-gnupg: Not Installed
          smmap: Not Installed
        timelib: Not Installed

System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-74-generic
         system: Ubuntu 14.04 trusty
```

**Comment by HeathNaylor** [comment]
Possibly related to https://github.com/saltstack/salt/issues/31454

**Comment by jfindlay** [comment]
@HeathNaylor, thanks for reporting.

**Comment by cachedout** [comment]
This is fixed at the head of the 2015.8 branch.

To test, I observed that `mine.update` increased memory for every invocation on the salt-minion on v2015.8.5. Using the head of the branch, however, this behavior does not occur.

Likely fixed by #32373
